compiler/quilt/nodes.py

```diff
@@ -98,19 +102,46 @@ def _class_repr(self):
         finfo = self._package.get_path()
         return "<%s %r>" % (self.__class__.__name__, finfo)
 
-    def _set(self, path, value):
+    def _set(self, path, value, build_dir=''):
+        """Create and set a node by path
+
+        This creates a node from a filename or pandas DataFrame.
+
+        If `value` is a filename, it must be relative to `build_dir`,
+            and it will be stored for export.
+            `build_dir` is the current directory by default.
+
+        :param path:  Path list -- I.e. ['examples', 'new_node']
+        :param value:  Pandas dataframe, or a filename relative to build_dir
+        :param build_dir:  Directory containing `value` if value is a filename.
+        """
         assert isinstance(path, list) and len(path) > 0
 
         if isinstance(value, pd.DataFrame):
-            core_node = core.TableNode(hashes=[], format=core.PackageFormat.default.value)
-        elif isinstance(value, string_types):
-            core_node = core.FileNode(hashes=[])
+            # all we really know at this point is that it's a pandas dataframe.
+            metadata = {'q_target': 'pandas'}
+            core_node = core.TableNode(hashes=[], format=core.PackageFormat.default.value, metadata=metadata)
+        elif isinstance(value, string_types + (bytes,)):
```

Comment:
When would `bytes` come up? That looks like a bug to me.

Reply:
We can opt not to take bytes.  My considerations were:
* `+` It's the same behaviour as `open()` and many other file operations in Python that accept names.
* `+` We have to consider 2.7, where `str is bytes`.
* `+` Including 'bytes' in py3 is consistent with what someone may do in in 2.7
* `+` `string_types` would still include bytes it in 2.7, so we'd still have to convert bytes over.
* `-` Perhaps we could use `bytes` for something else in the future, and this makes that option less friendly

Although that's what I figured, I have no problem rejecting bytes in 3.x (or altogether).

Reply:
Ah, that's fine then.

(I don't get why `open` accepts `bytes`... but oh well, might as well be consistent with that.)